asapdiscovery-docking/asapdiscovery/docking/scripts/single_target_docking.py

```diff
@@ -0,0 +1,416 @@
+        --debug \
+
+"""
+
+
+# setup input arguments
+parser = argparse.ArgumentParser(description="Run single target docking.")
+
+parser.add_argument(
+    "-r", "--receptor", required=True, help=("Path to receptor to prep and dock to")
+)
+
+parser.add_argument(
+    "-m",
+    "--mols",
+    required=True,
+    help=(
+        "Path to the molecules to dock to the receptor as an SDF or SMILES file, or SMILES string."
+    ),
+)
+
+parser.add_argument(
+    "--title",
+    help=(
+        "Title of molecule to use if a SMILES string is passed in as input, default is to hash the SMILES string to avoid accidental caching."
+    ),
+)
+
+parser.add_argument(
+    "-o",
+    "--output_dir",
+    required=True,
+    help="Path to output_dir, will NOT overwrite if exists.",
+)
+
+# general arguments
+parser.add_argument(
+    "--debug",
+    action="store_true",
+    help="enable debug mode, with more files saved and more verbose logging",
+)
+
+parser.add_argument(
+    "--verbose",
+    action="store_true",
+    help="Whether to print out verbose logging.",
+)
+
+# general arguments
+parser.add_argument(
+    "--cleanup",
+    action="store_true",
+    help="clean up intermediate files",
+)
+
+# Prep arguments
+parser.add_argument(
+    "--loop_db",
+    help="Path to loop database.",
+)
+parser.add_argument(
+    "--seqres_yaml",
+    help="Path to yaml file of SEQRES.",
+)
+parser.add_argument(
+    "--protein_only",
+    action="store_true",
+    default=False,
+    help="If true, generate design units with only the protein in them",
+)
+parser.add_argument(
+    "--ref_prot",
+    default=None,
+    type=str,
+    help="Path to reference pdb to align to. If None, no alignment will be performed",
+)
+
+# Docking arguments
+parser.add_argument(
+    "--top_n",
+    type=int,
+    default=1,
+    help="Number of top matches to dock. Set to -1 to dock all.",
+)
+parser.add_argument(
+    "--docking_sys",
+    default="posit",
+    help="Which docking system to use [posit, hybrid]. Defaults to posit.",
+)
+parser.add_argument(
+    "--relax",
+    default="none",
+    help="When to run relaxation [none, clash, all]. Defaults to none.",
+)
+
+parser.add_argument(
+    "--omega",
+    action="store_true",
+    help="Use Omega conformer enumeration.",
+)
+
+parser.add_argument(
+    "--hybrid",
+    action="store_true",
+    help="Whether to only use hybrid docking protocol in POSIT.",
+)
+
+parser.add_argument(
+    "--num_poses",
+    type=int,
+    default=1,
+    help="Number of poses to return from docking.",
+)
+
+parser.add_argument(
+    "--gat",
+    action="store_true",
+    help="Whether to use GAT model to score docked poses.",
+)
+
+
+def main():
+    args = parser.parse_args()
+
+    # setup output directory
+    output_dir = Path(args.output_dir)
+    output_dir.mkdir(parents=True, exist_ok=True)
+
+    logname = "prep-mcs-dock-single-target"
```

Comment:
make logname a script argument with a default name? useful for running docking in parallel (i.e. using job_arrays) to be able to distinguish the actions of the different jobs